src/applySchemaTyping.ts

```diff
@@ -0,0 +1,20 @@
+import type {LinkedJSONSchema} from './types/JSONSchema'
+import {typesOfSchema} from './typesOfSchema'
+
+export function applySchemaTyping(schema: LinkedJSONSchema) {
+  const types = typesOfSchema(schema)
+  schema.$types = types
+
+  if (types.length > 1) {
+    schema.$intersection = {
```

Comment:
`$intersection` and `$types` may conflict with schema properties (in JSON-Schema, users can define custom properties). Instead, can you use Symbols (similar to `Parent`)? This name collision could also be good to add as a test case.

https://github.com/bcherny/json-schema-to-typescript/blob/31993def993b610ba238d3024260129e31ddc371/src/types/JSONSchema.ts#L43

Reply:
This suggestion significantly improved the code quality, thank you.